randomteleport-plugin/src/main/java/de/themoep/randomteleport/RandomTeleport.java

```diff
@@ -364,4 +367,19 @@ public RandomSearcher runPreset(CommandSender sender, String preset, Player targ
         }
         return parseAndRun(sender, center, cmd.split(" "));
     }
+
+    @Override
+    public Location getRandomLocation(Player player, Location origin, int minRange, int maxRange, LocationValidator... validators) throws ExecutionException, InterruptedException {
+        return getRandomSearcher(player, origin, minRange, maxRange, validators).search().get();
+    }
+
+    @Override
+    public void teleportToRandomLocation(Player player, Location origin, int minRange, int maxRange, LocationValidator... validators) throws ExecutionException, InterruptedException {
+        player.teleport(getRandomLocation(player, origin, minRange, maxRange, validators));
```

Comment:
Please use the same teleport logic [used internally](https://github.com/Phoenix616/RandomTeleport/blob/master/randomteleport-plugin/src/main/java/de/themoep/randomteleport/RandomTeleport.java#L316) to ensure we will teleport efficiently and not block the thread this was called from. Should probably also return a CompletableFuture with information about whether or not the player was actually teleported.

Reply:
Also with the cooldown logic or only with the location y+1 ?

Reply:
No the cooldown would only apply to searchers that have a cooldown set in their settings. I guess I could actually check that there instead of adding entries without a cooldown to that table...

Regarding the y+1: I noticed that when not adding that there is a pretty reliable case of the player ending up in the ground so I used that to compensate for it. Seeing as this is meant as an API method I don't think we should add that internally though as the expectation would be that the player ends up at exactly these coordinates, not one higher so I wouldn't add that here.

I would still prefer if this returned a future indicating the result though as currently you have no idea if the player actually teleported or not. (Happens when no safe location was found)

Reply:
Changed it. Hopefully the CompletableFuture functions that way.